actors/common/src/main/java/org/sunbird/learner/actors/coursebatch/BatchOperationNotifierActor.java

```diff
@@ -0,0 +1,335 @@
+        removedParticipants.remove(entry.getKey());
+      }
+    }
+    courseBatchPrev.setParticipant(removedParticipants);
+    courseBatchNew.setParticipant(newParticipants);
+    courseBatchPrev.setMentors(new ArrayList<String>());
+    courseBatchNew.setMentors(new ArrayList<String>());
+    if (!removedParticipants.isEmpty())
+      batchCreateOperationNotifier(courseBatchPrev, JsonKey.REMOVE);
+    if (!newParticipants.isEmpty()) batchCreateOperationNotifier(courseBatchNew, JsonKey.ADD);
+  }
+
+  /*
+   * This method takes courseBatch object as parameter and returns a map which
+   * contains required data for sending email returns Map<String,Object>
+   */
+  private Map<String, Object> createRequestMap(CourseBatch courseBatch) {
+    Map<String, Object> courseBatchObject = new ObjectMapper().convertValue(courseBatch, Map.class);
+    Map<String, String> additionalCourseInfo =
+        (Map<String, String>) courseBatchObject.get(JsonKey.COURSE_ADDITIONAL_INFO);
+    Map<String, Object> requestMap = new HashMap<String, Object>();
+    requestMap.put(JsonKey.ORG_NAME, courseBatchObject.get(JsonKey.ORG_NAME));
+    requestMap.put(JsonKey.COURSE_LOGO_URL, additionalCourseInfo.get(JsonKey.COURSE_LOGO_URL));
+    requestMap.put(JsonKey.COURSE_NAME, additionalCourseInfo.get(JsonKey.COURSE_NAME));
+    requestMap.put(JsonKey.START_DATE, courseBatchObject.get(JsonKey.START_DATE));
+    requestMap.put(JsonKey.END_DATE, courseBatchObject.get(JsonKey.END_DATE));
+    requestMap.put(JsonKey.COURSE_ID, courseBatchObject.get(JsonKey.COURSE_ID));
+    requestMap.put(JsonKey.NAME, courseBatch.getName());
+    return requestMap;
+  }
+
+  private Map<String, String> getUserData(Map<String, Object> data, String operationType) {
+    Map<String, String> user = new HashMap<String, String>();
+    user.put(JsonKey.FIRST_NAME, data.get(JsonKey.FIRST_NAME).toString());
+    user.put(JsonKey.EMAIL, data.get(JsonKey.EMAIL).toString());
+    if (operationType.equalsIgnoreCase(JsonKey.ADD)) {
+      user.put(JsonKey.SUBJECT, JsonKey.COURSE_INVITATION);
+    } else if (operationType.equalsIgnoreCase(JsonKey.REMOVE)) {
+      user.put(JsonKey.SUBJECT, JsonKey.UNENROLL_FROM_COURSE);
+    }
+    return user;
+  }
+
+  /*
+   * This method is for sending e-mails for one at a time.
+   */
+  private void sendMail(Map<String, String> user, Map<String, Object> requestMap) {
+    String template = getEmailTemplateFile(user.get("templateName"));
+    requestMap.put(JsonKey.FIRST_NAME, user.get(JsonKey.FIRST_NAME));
+    VelocityContext context = getContext(requestMap);
+    String decryptedEmail = decryptionService.decryptData(user.get(JsonKey.EMAIL));
+    try {
+      SendMail.sendMailWithBody(
+          new String[] {decryptedEmail}, user.get(JsonKey.SUBJECT), context, template);
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "Error encountered while sending email notification for batch operation for user   "
```

Comment:
Logger message should be "ClassName:methodname Message , log label" .
put e.getMessage as well to track what error we are getting.